### PR TITLE
chore(deps): migrate to React

### LINKED ISSUE #61: Don't open up in a new window
When you try to sign in, the sign in flow opens up in a new window. This feels a bit unnecessary given how little information we need to input. Can't we instead do this in the popover dialogue from the bar?

If we can't, then we at least should auto-close this page:
Screenshot 2024-12-03 at 1 54 38 PM

### LINKED ISSUE #63: Fix documentation, flow and errror
I was trying to add a YouTube Studio dashboard to my screens:

Screenshot 2024-12-03 at 1 55 47 PM

However, when I press save, I get this:
Screenshot 2024-12-03 at 1 55 56 PM

Unclear what the error is an how to debug it.

Also, please rewrite the error message under 'save authentication' and fix the formatting there.

### PR DESCRIPTION
**Pull request by nicomiguelino**
### Checklist

- [x] Resolve code review comments.
- [ ] Include changes that resolve #61, if that gets merged first.
- [ ] Include changes that resolve #63, if that gets merged first.

### Overview

* An alternative pull request to #42
* We decided to stick to [React](https://react.dev/), as it's a more established framework with an active community.
* Artifacts can be found here: https://github.com/Screenly/Browser-Extension/actions/runs/12154319850?pr=47

image

### Extension Files (For Testing)

* https://github.com/Screenly/Chrome-Extension/actions/runs/12036958215/artifacts/2240613970

### Screenshots

image

image

image

### Screencasts

[01-sign-in-error.webm](https://github.com/user-attachments/assets/0c1f6d24-b189-4f8e-b667-ba9ba2c2c8ae)

[02-sign-in-success.webm](https://github.com/user-attachments/assets/49d5c335-a363-4dcb-8b40-14c7494d5598)

[03-sign-out.webm](https://github.com/user-attachments/assets/ae813d48-266d-40ac-beb8-742c4cde58b6)

[04-add-asset.webm](https://github.com/user-attachments/assets/05c96cdb-5f77-4095-8767-4909b90ddc65)

[05-update-asset.webm](https://github.com/user-attachments/assets/a384b76c-7155-48fe-9886-5a96ebcd9de8)

### CODE REVIEW

**Review by vpetersson — APPROVED** [review]
Should probably be reviewed by @iamplugged or @rusko124

**Comment by nicomiguelino** [comment]
@rusko124, it looks like the teams endpoint returns two entries, when it's supposed to return only one, as per @vpetersson:

image

**Review by vpetersson — COMMENTED** [review]
Please improve the error messages.

File: src/assets/js/boot-options.mjs
Comment (iamplugged) [inline src/assets/js/boot-options.mjs:21]: Component name could probably be better to reflect what it does

File: src/assets/js/boot-options.mjs
Comment (iamplugged) [inline src/assets/js/boot-options.mjs:35]: For attributes within tags, prefer using double quotes everywhere. For any JS stuff use single quotes.

File: src/assets/js/boot-options.mjs
Comment (iamplugged) [inline src/assets/js/boot-options.mjs:16]: For js/jsx files, you can optionally avoid adding extensions.

import { SignIn } from '@/components/options/sign-in';

should be good enough.

File: src/assets/js/components/options/sign-in.jsx
Comment (iamplugged) [inline src/assets/js/components/options/sign-in.jsx:1]: Directory name `options` could be better

File: src/assets/js/components/options/sign-in.jsx
Comment (iamplugged) [inline src/assets/js/components/options/sign-in.jsx:48]: return might not be required here

File: src/assets/js/components/options/sign-in.jsx
Comment (iamplugged) [inline src/assets/js/components/options/sign-in.jsx:53]: If setIsLoading(false); is already set in finally block, it is probably unnecessary here

File: src/assets/js/components/options/sign-in.jsx
Comment (iamplugged) [inline src/assets/js/components/options/sign-in.jsx:59]: The return is already wrapped inside a <div>.

You can either use <></> or a parent div to return the JSX

File: src/assets/js/components/popup/loading.jsx
Comment (iamplugged) [inline src/assets/js/components/popup/loading.jsx:3]: Can probably be better named as PopupSpinner or OverlaySpinner

File: src/assets/js/components/popup/proposal.jsx
Comment (iamplugged) [inline src/assets/js/components/popup/proposal.jsx:33]: Needs better messaging

File: src/assets/js/features/popup/popupSlice.js
Comment (iamplugged) [inline src/assets/js/features/popup/popupSlice.js:10]: States like loading, success should ideally be associated to a particular action/side effect.

For example for a sign in action, it should have loading, success, fail states linked to it.

File: src/assets/js/components/popup/success.jsx
Comment (iamplugged) [inline src/assets/js/components/popup/success.jsx:38]: Better naming for the id

File: src/assets/js/components/popup/save-auth-warning.jsx
Comment (iamplugged) [inline src/assets/js/components/popup/save-auth-warning.jsx:9]:
```suggestion
  useEffect(() => {
    setHostname(props.hostname);
  }, [props.hostname]);
```

File: src/assets/js/components/popup/proposal.jsx
Comment (iamplugged) [inline src/assets/js/components/popup/proposal.jsx:260]: Remove logs

File: src/assets/js/components/popup/proposal.jsx
Comment (iamplugged) [inline src/assets/js/components/popup/proposal.jsx:226]: Remove log

**Review by iamplugged — CHANGES_REQUESTED** [review]
General - Use better naming for component names, ids, variables etc.

File: src/assets/js/components/options/sign-in.jsx
Comment (nicomiguelino) [reply src/assets/js/components/options/sign-in.jsx:1]: @iamplugged, a browser extension can be composed of an [options page](https://developer.chrome.com/docs/extensions/reference/manifest) and a [popup page](https://developer.chrome.com/docs/extensions/develop/ui/add-popup), so I decided to rename it to `options`.

**Comment by nicomiguelino** [comment]
@vpetersson, I already enabled ESLint in JS and JSX files a few commits ago.

**Review by vpetersson — APPROVED** [review]
Approved pending @iamplugged's approval.

File: src/assets/js/components/popup/proposal.jsx
Comment (iamplugged) [inline src/assets/js/components/popup/proposal.jsx:186]: It will be undefined by default, even if you don't explicitly initialize to undefined

File: src/assets/js/components/popup/proposal.jsx
Comment (iamplugged) [inline src/assets/js/components/popup/proposal.jsx:197]: Same here

File: src/assets/js/main.mjs
Comment (iamplugged) [inline src/assets/js/main.mjs:7]: data, token would be undefined by default even if nit initialized explicitly

**Review by iamplugged — APPROVED** [review]
Approved with minor comments

**Comment by nicomiguelino** [comment]
@iamplugged, I'll be resolving the minor comments in a separate pull request.

File: src/assets/js/components/popup/proposal.jsx
Comment (nicomiguelino) [reply src/assets/js/components/popup/proposal.jsx:186]: A declaration is still needed. I can just do `let headers` or maybe `let headers = {}`

**Pull request merged**
